src/helpers.rkt

```diff
@@ -9,9 +9,9 @@
 
 (define (is-arrow-type? expr)
   (and (list? expr)
-       (equal? (first expr) '->)))
+       (eq? (first expr) '->)))
```

Comment:
So, where would this affect me? From the docs I understand `equal?` as "`eq?` unless defined otherwise for the particular datatype"